### PR TITLE
Docs: adding error rate spike use case for MCP

### PR DESCRIPTION
**Pull request by gkarthi-signoz**
## Summary
Adds a new "Error Rate Spike Explainer" guide to the AI use cases documentation showing how to investigate error rate spikes using SigNoz MCP server.

## Motivation / Problem
Users need practical guidance on how to use the SigNoz AI/MCP integration to debug error rate spikes by tracing errors through the call chain to identify the root cause (whether in their service or downstream dependencies).

## Changes
- Added new doc: [data/docs/ai/use-cases/error-rate-spike-explainer.mdx](data/docs/ai/use-cases/error-rate-spike-explainer.mdx)
- Updated sidebar navigation in [constants/docsSideNav.ts](constants/docsSideNav.ts#L2963-L2968) to include the new guide
- Updated AI use cases index page [data/docs/ai/use-cases.mdx](data/docs/ai/use-cases.mdx) to reference the new guide

## Description
This PR adds a comprehensive step-by-step guide demonstrating how to investigate an error rate spike using AI assistant connected to SigNoz via MCP. The guide walks through a realistic scenario: grouping errors by operation, analyzing trace call chains, and identifying Postgres connection pool exhaustion as the root cause.

## Type of Change

- [x] **Docs** – Changes to `data/docs/**`
- [ ] **Blog** – Changes to `data/blog/**`
- [ ] **Site Code** – Changes to `app/**`, `components/**`, `hooks/**`, `utils/**`, config, etc.
- [ ] **Redirects** – Renamed/moved docs or updated `next.config.js` redirects
- [ ] **Dependencies / CI / Scripts**
- [ ] **Other** – Explain below

## Impact

- [x] Documentation only
- [ ] UI changes
- [ ] Developer workflow
- [ ] Performance impact
- [ ] Breaking change

If breaking change, describe migration steps:

### General
- [x] Branch is up to date with `main`
- [x] PR title follows conventional format (`type: description`)
- [x] Self-reviewed the code
- [ ] Comments added for complex logic

### For all changes
- [x] Built locally (`yarn build`) with no errors
- [x] Ran `yarn lint` and fixed any issues
- [x] Pre-commit hooks passed (or ran `yarn check:doc-redirects` / `yarn check:docs-metadata` if applicable)

### For docs changes (`data/docs/**`)
- [ ] Followed the docs author checklist in [contributing/docs-authoring.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/docs-authoring.md)
- [ ] Added/updated the page in `constants/docsSideNav.ts` if adding or moving a doc
- [ ] Any added docs images use WebP format and live under `public/img/docs/<topic>/`

### For blog changes
- [ ] Followed the blog workflow in [contributing/blog-workflow.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/blog-workflow.md)
- [ ] Frontmatter includes `title`, `date`, `author`, `tags` (and `canonicalUrl` if applicable)
- [ ] Images use WebP format and live under `public/img/blog/<YYYY-MM>/`

### For site code changes
- [ ] Followed the site code playbook in [contributing/site-code.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/site-code.md)
- [ ] New dependencies are justified in the PR description (if any)

### For renamed or moved docs
- [ ] Followed the redirects and discovery section in [contributing/docs-authoring.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/docs-authoring.md)
- [ ] Added permanent redirect in `next.config.js` under `async redirects()`
- [ ] Updated internal links and sidebar in `constants/docsSideNav.ts`
- [ ] Ran `yarn check:doc-redirects` to verify

## Testing

- [x] Tested locally
- [ ] Edge cases considered
- [ ] Existing functionality verified
- [ ] New tests added (if applicable)

Steps to test:
1.
2.
3.

## Additional Context

---

**Note:** Submit as Draft by default. Mark "Ready for review" when checks pass and content is ready.

### CODE REVIEW

File: data/docs/ai/use-cases/error-rate-spike-explainer.mdx [hunk 1]
```diff
@@ -0,0 +1,206 @@
+---
+date: 2026-04-07
+id: error-rate-spike-explainer
+title: Error Rate Spike Explainer
+description: Find where errors originate in the call chain when error rates spike on a service.
+doc_type: howto
+---
```
[the remark this answers is not in the document]
Comment (gkarthi-signoz) [reply data/docs/ai/use-cases/error-rate-spike-explainer.mdx:7]: other docs are howto for mcp tools

File: data/docs/ai/use-cases/error-rate-spike-explainer.mdx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+Your alert fires: `frontend error rate >= 10%, triggered 6 min ago`. You know something is broken. You need to know where.
+
+Is the bug in your service, or is a downstream dependency failing and propagating errors upstream? You open your AI assistant, connected to SigNoz via the MCP server, and start investigating.
+
+## Prerequisites
+
+- Connect your AI assistant to SigNoz using the [MCP Server guide](https://signoz.io/docs/ai/signoz-mcp-server).
+- Make sure your services are instrumented with distributed tracing. See [Instrument Your Application](https://signoz.io/docs/instrumentation/) if you haven't set this up.
+
+## Step 1: Group By Error
```
[the remark this answers is not in the document]
Comment (gkarthi-signoz) [reply data/docs/ai/use-cases/error-rate-spike-explainer.mdx:20]: no it ac works

**Review by makeavish — APPROVED** [review]
